Wikipedia/Code/ShiftingTopViewsContaining.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+import SwiftUI
+
+protocol ShiftingTopViewsContaining: UIViewController {
+    var shiftingTopViewsStack: ShiftingTopViewsStack? { get set }
+}
+
+// Note: This subclass only seems necessary for proper nav bar hiding in iOS 14 & 15. It can be removed and switched to raw UIHostingControllers for iOS16+
+private class NavigationBarHidingHostingVC<Content: View>: UIHostingController<Content> {
```

Comment:
It would be good to prevent the bar from hiding when VoiceOver is on or provide another way for VoiceOver users to go back from this view when it is hiding (This can be done on the part 2 PR).

Reply:
Great catch, I hadn't thought about Voice Over yet. 😅 I'll spin it off into a Part 3 PR since Part 2 was getting pretty big.